plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -395,6 +482,50 @@ private ParquetFileWriter createParquetFileWriter(Location path, List<DeltaLakeC
         }
     }
 
+    private boolean isNoRowsRetained(String path, Roaring64NavigableMap pastDeletionVectors, FileDeletion deletion)
+    {
+        LongBitmapDataProvider rowsDeletedByDelete = deletion.rowsDeletedByDelete();
+        LongBitmapDataProvider rowsDeletedByUpdate = deletion.rowsDeletedByUpdate();
+        int retainedCount = 0;
+        try (ConnectorPageSource connectorPageSource = createParquetPageSource(Location.of(path)).get()) {
+            long filePosition = 0;
+            while (!connectorPageSource.isFinished()) {
+                Page page = connectorPageSource.getNextPage();
+                if (page == null) {
+                    continue;
+                }
+
+                int positionCount = page.getPositionCount();
```

Comment:
We don't seem to read anything from the Page, could you get the filePosition by just knowing the overall file positions count from the file footer ?